src/main/java/minijava/ir/utils/FirmUtils.java

```diff
@@ -0,0 +1,46 @@
+package minijava.ir.utils;
+
+import firm.nodes.Address;
+import firm.nodes.Block;
+import firm.nodes.Node;
+import firm.nodes.Phi;
+import java.util.*;
+
+public class FirmUtils {
+
+  private static Map<Phi, Boolean> isPhiProneToLostCopies = new HashMap<>();
+
+  public static String getMethodLdName(firm.nodes.Call node) {
+    return ((Address) node.getPred(1)).getEntity().getLdName();
+  }
+
+  public static boolean isPhiProneToLostCopies(Phi phi) {
+    if (!isPhiProneToLostCopies.containsKey(phi)) {
+      isPhiProneToLostCopies.put(phi, checkPhi(phi));
+    }
+    return isPhiProneToLostCopies.get(phi);
+  }
```

Comment:
Why? `checkPhi` is costly as we traverse the whole graph and the methods is used quite often (several times on each phi node).